openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -517,3 +562,116 @@ def partial_update(self, request):
             return Response(text_type(exc), status=status.HTTP_400_BAD_REQUEST)
         except Exception as exc:  # pylint: disable=broad-except
             return Response(text_type(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class AccountRetirementView(ViewSet):
+    """
+    Provides API endpoint for retiring a user.
+    """
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, CanRetireUser,)
+    parser_classes = (JSONParser,)
+
+    @request_requires_username
+    def post(self, request):
+        """
+        POST /api/user/v1/accounts/retire/
+
+        {
+            'username': 'user_to_retire'
+        }
+
+        Retires the user with the given username.  This includes
+        retiring this username, the associates email address, and
+        any other PII associated with this user.
+        """
+        username = request.data['username']
+        if is_username_retired(username):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            user = UserRetirementStatus.get_retirement_for_retirement_action(username).user
+
+            with transaction.atomic():
```

Comment:
This isn't really needed here as the view itself will be atomic by default. This should just create a savepoint within the top level transaction. Won't hurt anything, but may be misleading to read.

Reply:
Ah, good point.